go/libraries/doltcore/sqle/database.go

```diff
@@ -117,56 +125,114 @@ func (db *Database) GetTableInsensitive(ctx context.Context, tblName string) (sq
 		return NewLogTable(db.ddb, db.rsr), true, nil
 	}
 
-	tableNames, err := db.GetAllTableNames(ctx)
+	return db.getTable(ctx, db.root, tblName)
+}
 
+// GetTableInsensitiveAsOf implements sql.VersionedDatabase
+func (db *Database) GetTableInsensitiveAsOf(ctx *sql.Context, tableName string, time interface{}) (sql.Table, bool, error) {
+	root, err := db.rootAsOf(ctx, time)
 	if err != nil {
 		return nil, false, err
 	}
 
-	exactName, ok := sql.GetTableNameInsensitive(tblName, tableNames)
+	return db.getTable(ctx, root, tableName)
+}
 
+func (db *Database) rootAsOf(ctx *sql.Context, time interface{}) (*doltdb.RootValue, error) {
```

Comment:
don't love time as a variable name as it's the name of a standard go package.

Reply:
Fixed.